src/chainstate/stacks/db/blocks.rs

```diff
@@ -5143,6 +5205,18 @@ impl StacksChainState {
         let evaluated_epoch = clarity_tx.get_epoch();
         clarity_tx.reset_cost(parent_block_cost.clone());
 
+        debug!("Evaluating block with epoch = {}", evaluated_epoch);
+        if evaluated_epoch >= StacksEpochId::Epoch21 {
+            Self::check_and_handle_reward_start(
```

Comment:
Oh interesting -- this is happening before any transactions get processed in this block, including those in the parent microblock stream.  Is that what we want?  Asking because the parent microblock stream could have been mined at any point in the past (including an ancient reward cycle), given that the anchored block miner can confirm any microblock stream they want.  Should this be moved to after the microblock transaction processing happens, but before the anchored block transaction processing happens?

Reply:
No, I think this should happen after the parent microblocks.